sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala

```diff
@@ -121,6 +123,11 @@ private[hive] object SparkSQLCLIDriver extends Logging {
       }
     }
 
+    Option(new HiveDelegationTokenProvider)
+      .filter(_.delegationTokensRequired(sparkConf, hadoopConf))
+      .foreach(_.obtainDelegationTokens(
+        hadoopConf, sparkConf, UserGroupInformation.getCurrentUser.getCredentials))
```

Comment:
This will not insert the tokens into the current UGI, because `getCredentials` returns a copy.

Reply:
OK，i got it